app/src/test/java/nerd/tuxmobil/fahrplan/congress/repositories/LectureListTransformerTest.kt

```diff
@@ -0,0 +1,181 @@
+                createLecture(lectureId = "L0", room = "Three", roomIndex = 12),
+                createLecture(lectureId = "L1", room = "Two", roomIndex = 11),
+                createLecture(lectureId = "L2", room = "One", roomIndex = 10)
+        )
+
+        val result = transformer.legacyTransformLectureList(dayIndex = 1, lectures = lecturesInDatabase)
+
+        with(result.roomsMap) {
+            assertThat(get("One")).isEqualTo(0)
+            assertThat(get("Two")).isEqualTo(1)
+            assertThat(get("Three")).isEqualTo(2)
+        }
+    }
+
+
+    @Test
+    fun `roomList maps to rooms in correct order`() {
+        val lecturesInDatabase = listOf(
+                createLecture(lectureId = "L0", room = "Three", roomIndex = 12),
+                createLecture(lectureId = "L1", room = "Two", roomIndex = 11),
+                createLecture(lectureId = "L2", room = "One", roomIndex = 10)
+        )
+
+        val result = transformer.legacyTransformLectureList(dayIndex = 1, lectures = lecturesInDatabase)
+
+        val firstRoomIndex = result.roomList.get(0)
+        val secondRoomIndex = result.roomList.get(1)
+        val thirdRoomIndex = result.roomList.get(2)
+        val roomIndexToRoomNameMap = result.roomsMap.entries.associate{ (key, value) -> value to key }
+        assertThat(roomIndexToRoomNameMap[firstRoomIndex]).isEqualTo("One")
+        assertThat(roomIndexToRoomNameMap[secondRoomIndex]).isEqualTo("Two")
+        assertThat(roomIndexToRoomNameMap[thirdRoomIndex]).isEqualTo("Three")
+    }
+
+    @Test
+    fun `Lecture_roomIndex is updated with rewritten roomIndex`() {
+        val lecturesInDatabase = listOf(
+                createLecture(lectureId = "L0", room = "Three", roomIndex = 12),
+                createLecture(lectureId = "L1", room = "Two", roomIndex = 11),
+                createLecture(lectureId = "L2", room = "One", roomIndex = 10)
+        )
+
+        val result = transformer.legacyTransformLectureList(dayIndex = 1, lectures = lecturesInDatabase)
+
+        assertThat(result.lectureList.first { it.room == "One" }.roomIndex).isEqualTo(0)
+        assertThat(result.lectureList.first { it.room == "Two" }.roomIndex).isEqualTo(1)
+        assertThat(result.lectureList.first { it.room == "Three" }.roomIndex).isEqualTo(2)
+    }
+
+    @Test
+    fun `roomCount matches number of rooms`() {
+        val lecturesInDatabase = listOf(
+                createLecture(lectureId = "L0", room = "One", roomIndex = 10),
+                createLecture(lectureId = "L1", room = "Two", roomIndex = 11)
+        )
+
+        val result = transformer.legacyTransformLectureList(dayIndex = 1, lectures = lecturesInDatabase)
+
+        assertThat(result.roomCount).isEqualTo(2)
+    }
+
+    private fun createLecture(lectureId: String, room: String, roomIndex: Int): Lecture {
+        return Lecture(lectureId).apply {
+            this@apply.room = room
+            this@apply.roomIndex = roomIndex
+        }
+    }
+}
```

Comment:
The `@apply` references are not needed.

Can you please rename `room` to `roomName` here, too?

The method could be shortened by using the expression body syntax.

Reply:
That'll either make for a very long line or weird indentation when the `Lecture(lectureId)` is put on a new line. What's your preference?

Reply:
I go with readability. I leave it up to you to decide. The expression body syntax was just meant as an option.